apps/gateway/lib/gateway/controllers/curse_of_mirra/item_controller.ex

```diff
@@ -20,13 +21,17 @@ defmodule Gateway.Controllers.CurseOfMirra.ItemController do
 
   # This is a placeholder and will be developed properly in https://github.com/lambdaclass/mirra_backend/issues/638
   def buy(conn, params) do
-    with {:ok, item_template_id} <-
-           Items.get_purchasable_template_id_by_name_and_game_id(
+    with {:ok, item_template} <-
+           Items.get_purchasable_template_by_name_and_game_id(
              params["item_name"],
              Utils.get_game_id(:curse_of_mirra)
            ),
-         {:ok, item} <- Items.insert_item(%{user_id: params["user_id"], template_id: item_template_id}) do
-      send_resp(conn, 200, Jason.encode!(item.id))
+         {:ok, currency} <-
+           Currencies.get_currency_by_name_and_game(params["currency_name"], Utils.get_game_id(:curse_of_mirra)),
+         {:ok, purchase_cost} <- Items.get_purchase_cost_by_currency(currency.id, item_template),
+         {:can_afford, true} <- {:can_afford, Currencies.can_afford(params["user_id"], [purchase_cost])},
+         {:ok, item_updates_map} <- Items.buy_item(params["user_id"], item_template.id, [purchase_cost]) do
```

Comment:
The issue says:
```
Allow ItemTemplate to have more than one ItemCost (has_many assoc).
ItemCost will have an assoc to more than one CurrencyCost (embeds_many assoc).
```

But that's not true here, we have a list of costs and if we try to buy it with a currency and the template has that cost and we can afford it we bough it, no composite price is involved. This is to simplify implementation? can we update the issue?

Reply:
Sure. If you see the commits history it was introduced in this PR but I deleted it after debating with Marco and Lotu.
You can check Lotu's comments too.